src/modules/lua_mtev.c

```diff
@@ -1094,6 +1107,42 @@ mtev_lua_socket_read_complete(eventer_t e, int mask, void *vcl,
   return 0;
 }
 
+static int on_timeout(eventer_t e, int mask, void *closure,
+    struct timeval *now) {
+  struct nl_slcl *cl;
+    mtev_lua_resume_info_t *ci;
+  lua_timeout_callback_ref* cb_ref;
+  lua_State *L;
+
+  cb_ref = (lua_timeout_callback_ref*)closure;
+  L = cb_ref->L;
+
+  // run the timeout callback
+  lua_rawgeti( L, LUA_REGISTRYINDEX, cb_ref->callback_reference );
+  lua_call(L, 0, 0);
+
+  cl = cb_ref->timed_out_eventer->closure;
+  ci = mtev_lua_get_resume_info(L);
+  assert(ci);
+
+  // remove the original read event
+  eventer_remove_fd(cb_ref->timed_out_eventer->fd);
+  mtev_lua_deregister_event(ci, cb_ref->timed_out_eventer, 0);
+
+  // resume the original e:read() call and return nil
```

Comment:
This comment is misleading.  on_timeout will likely be used for other timeout socket ops... 

The subsequent four lines of code are part of the removal above... make them a contiguous block of 6 lines.

put this comment right before the lua_pushnil(L) and say something like...

// return into lua wherever we might have been blocking